low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -269,6 +273,68 @@ def prepare_mem_json(self, status, description):
         }
         return memory_dict
 
+    def get_nw_ports_info(self):
+        """Return the Network ports information."""
+        network_cable_data = []
+        io_counters = psutil.net_io_counters(pernic=True)
+
+        for interface, addrs in psutil.net_if_addrs().items():
+            nic_info = self.get_health_template(interface, False)
+
+            specifics = {}
+            for addr in addrs:
+                if addr.family == AF_INET:
+                    specifics["ipV4"] = addr.address
+
+            if interface in io_counters:
+                io_info = io_counters[interface]
+                specifics = {
+                    "networkErrors": io_info.errin + io_info.errout,
+                    "droppedPacketsIn": io_info.dropin,
+                    "droppedPacketsOut": io_info.dropout,
+                    "packetsIn": io_info.packets_recv,
+                    "packetsOut": io_info.packets_sent,
+                    "trafficIn": io_info.bytes_recv,
+                    "trafficOut": io_info.bytes_sent
+                }
+
+            nw_status, nw_cable_conn_status = \
+                self.get_nw_status(interface)
+
+            specifics["nwStatus"] = nw_status
+            specifics["nwCableConnStatus"] = nw_cable_conn_status
+
+            # set health status and description
+            map_status = {"UP": "OK", "DOWN": "Disabled/Failed",
+                          "UNKNOWN": "NA"}
+            health_status = map_status[nw_cable_conn_status]
+
+            desc = "Network Interface '%s' is %sin good state" % (
+                interface, '' if health_status == "OK" else 'not '
+            )
+
+            self.set_health_data(nic_info, health_status, description=desc,
+                                 specifics=[specifics])
+
+            network_cable_data.append(nic_info)
+
+        return network_cable_data
+
+    def get_nw_status(self, interface):
+        """Read & Return the latest network status from sysfs files."""
+        try:
+            nw_status = self.nw_instance.get_operational_state(interface)
+        except Exception:
+            nw_status = "UNKNOWN"
+            # Log the error when logging class is in place.
+        try:
+            nw_cable_conn_status = self.nw_instance.get_link_state(interface)
+        except Exception:
+            nw_cable_conn_status = "UNKNOWN"
+            # Log the error when logging class is in place.
+
+        return nw_status, nw_cable_conn_status
+
 
 # if __name__ == "__main__":
 #     server = ServerMap()
```

Comment:
Ensure regression test on this module is done. Once done, comment the code.
Use below as reference.
```
# if __name__ == "__main__":
#     server = ServerMap()
#     for fru in server.server_frus:
#         print(f"\nFRU: {fru}")
#         health_data = server.get_health_info(rpath=f"nodes[0]>compute[0]>hw>{fru}")
#         print(health_data)
```

Reply:
Yes, @mariyappanp test it, just didn't included the change in PR.